src/redis-cli.c

```diff
@@ -1960,6 +1963,15 @@ static int parseOptions(int argc, char **argv) {
             config.stdin_tag_name = argv[++i];
         } else if (!strcmp(argv[i],"-p") && !lastarg) {
             config.conn_info.hostport = atoi(argv[++i]);
+        } else if (!strcmp(argv[i],"-t") && !lastarg) {
+            double seconds = atof(argv[++i]);
+            long long milliseconds = seconds * 1000;
+            if (milliseconds < 0) {
+                fprintf(stderr, "Invalid connection timeout for -t.\n");
+                exit(1);
+            }
+            config.connect_timeout.tv_sec = milliseconds / 1000;
+            config.connect_timeout.tv_usec = (milliseconds % 1000) * 1000;
```

Comment:
Not sure why you need to go through milliseconds.
```suggestion
            double seconds = atof(argv[++i]);
            if (seconds < 0) {
                fprintf(stderr, "Invalid connection timeout for -t.\n");
                exit(1);
            }
            config.connect_timeout.tv_sec = (long long)seconds;
            config.connect_timeout.tv_usec = (((long long)seconds) * 1000000) % 1000000;
```

Also perhaps try to use `strtod` for input error checking instead of `atof`. This isn't critical, but we need at some point to fix all input checking in _redis-cli_ accordingly.

Reply:
right, silly me, i must be doing something wrong in the switching between milliseconds and seconds, thanks

> config.connect_timeout.tv_usec = (((long long)seconds) * 1000000) % 1000000;

this should be `config.connect_timeout.tv_usec = (long long)(seconds * 1000000) % 1000000;`?

Reply:
No, cast seconds to long long before multiplication. Otherwise, the multiplication is done with ints and it may overflow. I think.

Reply:
seconds is a double. like if seconds is 0.1, cast seconds to long long before multiplication, it will become 0?
I may need to change to: `config.connect_timeout.tv_usec = ((long long)(seconds * 1000000)) % 1000000;` to be clear

Reply:
You're right, sorry.